### ISSUE TITLE
Gunmods require you can fit both weapon and mod in pockets

### ISSUE THREAD

**Issue by NetSysFire** [CleverRaven/Cataclysm-DDA#69423]
### Describe the bug

There is no way to attach e.g a wooden arrow rest to a short bow when naked or with only unsuitable pockets.

### Attach save file

n/a

### Steps to reproduce

1. Be naked.
2. Spawn a short bow and wear it.
3. Spawn a wooden arrow rest and wield it.
4. Apply the wooden arrow rest, select your short bow and press enter to accept.
5. Stop wielding arrow rest? Only option is to drop it.
6. You no longer have a wooden arrow rest and can not continue crafting.

### Expected behavior

So far all the mods are small enough that you can simply drop them on the ground and work with them there, provided you are neither flying nor standing on rubble or something else.

Plus, it does not even attempt to use a nearby crafting spot or workbench and you can not resume crafting, it immediately reverts to your normal short bow.

### Versions and configuration

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: Arch; Description: Arch Linux; Release: rolling; Codename: n/a; 
- Game Version: cc4cb54 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

I could have sworn I reported this before but I can not find any issues about this.

**Comment by Vaskritaya** [comment]
Yeah, this is really fiddly and annoying to deal with. Ideally it would just work if the mod and target gun are at least adjacent to you, like how most similar mechanics work.

**Comment by Terrorforge** [comment]
Needing to wield the weapon is in large part a solution to a problem where it was otherwise possible to make e.g. a gun in a holster too large to fit in that holster, causing problems when it's drawn.

But I don't see any particular reason the mod _has_ to be in your inventory.

**Comment by Bobtron** [comment]
Agreed, the mod shouldn't need to be on the character. 
Possible solution is to add the flag `"ALLOWS_REMOTE_USE"` to all gun mods, that way they can be 'activated' in the activate menu even if not wielded. 
Then inside `void Character::gunmod_add(` in character_guns.cpp, rather than use `std::vector<item *> mods = items_with(...`, we could use `crafting_inventory()`, and fetch the `std::vector<item *> mods` from the 'crafting' inventory instead of the personal inventory, that way the mod can still be added even if dropped, when the gun is wielded. This seems a bit hacky though.
